File: app/src/main/java/app/revanced/bilibili/patches/DarkSwitchPatch.java
package app.revanced.bilibili.patches;

import android.app.AlertDialog;

import androidx.annotation.Keep;

import app.revanced.bilibili.settings.Settings;
import app.revanced.bilibili.utils.Utils;
import tv.danmaku.bili.ui.main2.mine.HomeUserCenterFragment;
import tv.danmaku.bilibilihd.ui.main.mine.HdHomeUserCenterFragment;

@Keep
public class DarkSwitchPatch {
    public static void switchDarkMode(HomeUserCenterFragment fragment, boolean report) {
        if (!Settings.SWITCH_DARK_DIALOG.getBoolean()) {
            fragment.switchDarkMode_Origin(report);
            return;
        }
        if (Utils.isNightFollowSystem()) {
            new AlertDialog.Builder(fragment.getContext())
                    .setMessage(Utils.getString("biliroaming_switch_dark_tips"))
                    .setPositiveButton(android.R.string.ok, (dialog, which) -> fragment.switchDarkMode_Origin(report))
                    .setNegativeButton(android.R.string.cancel, null)
                    .show();
        } else {
            fragment.switchDarkMode_Origin(report);
        }
    }

    public static void switchDarkMode(HdHomeUserCenterFragment fragment, boolean report) {
        if (!Settings.SWITCH_DARK_DIALOG.getBoolean()) {
            fragment.switchDarkMode_Origin(report);
            return;
        }
        if (Utils.isNightFollowSystem()) {
            new AlertDialog.Builder(fragment.getContext())
                    .setMessage(Utils.getString("biliroaming_switch_dark_tips"))
                    .setPositiveButton(android.R.string.ok, (dialog, which) -> fragment.switchDarkMode_Origin(report))
                    .setNegativeButton(android.R.string.cancel, null)
                    .show();
        } else {
            fragment.switchDarkMode_Origin(report);
        }
    }
}

File: integrations/app/src/main/java/app/revanced/bilibili/patches/DarkSwitchPatch.kt
package app.revanced.bilibili.patches

import android.app.AlertDialog
import androidx.annotation.Keep
import androidx.fragment.app.Fragment
import app.revanced.bilibili.settings.Settings
import app.revanced.bilibili.utils.Utils
import app.revanced.bilibili.utils.constraintSize
import app.revanced.bilibili.widget.OnSwitchDarkModeOriginListener

object DarkSwitchPatch {
    @Keep
    @JvmStatic
    fun switchDarkMode(listener: OnSwitchDarkModeOriginListener, report: Boolean) {
        if (!Settings.SwitchDarkTipsDialog()) {
            listener.switchDarkMode_Origin(report)
            return
        }
        if (Utils.isNightFollowSystem()) {
            val context = (listener as Fragment).context
            AlertDialog.Builder(context)
                .setMessage(Utils.getString("biliroaming_switch_dark_tips"))
                .setPositiveButton(android.R.string.ok) { _, _ ->
                    listener.switchDarkMode_Origin(report)
                }
                .setNegativeButton(android.R.string.cancel, null)
                .create().constraintSize().show()
        } else {
            listener.switchDarkMode_Origin(report)
        }
    }
}
